txpool/txpool.go

```diff
@@ -589,20 +589,9 @@ func (p *TxPool) addTx(origin txOrigin, tx *types.Transaction) error {
 
 	tx.ComputeHash()
 
-	// check if already known
-	if _, ok := p.index.get(tx.Hash); ok {
-		if origin == gossip {
-			// silently drop known tx
-			// that is gossiped back
-			p.logger.Debug(
-				"dropping known gossiped transaction",
-				"hash", tx.Hash.String(),
-			)
-
-			return nil
-		} else {
-			return ErrAlreadyKnown
-		}
+	//	add to index
+	if ok := p.index.add(tx); !ok {
+		return ErrAlreadyKnown
 	}
```

Comment:
I think you can move this check above so that we can skip transaction validation twice. What do you think?
cc: @zivkovicmilos

Reply:
Sounds good, as long as the transaction is removed from the index map, if triggering overflow and validate errors. One thing to note about the hash used for the index - I remember the old implementation commenting on the hash can be computed after all `validateTx` checks have passed.

Not sure what it meant at the time, but if it's okay to compute the hash of the transaction before validation checks, I'll move the indexing up top of the call stack. (cc @zivkovicmilos )

Reply:
@dbrajovic 

The comment was referring to the fact that we were initializing the `From` field from the `V`, `R`, `S` sig fields.
That way computing the hash would be unique, as the `From` field would be initialized

Reply:
So it's perfectly fine to compute the hash before validation ?